경북대 BE_배준호 4주차 과제 (1단계)

BaeJunH0 opened the pull request:
테스트에 대한 부분 외에 요구사항 충족한 것 같아서 PR 보냅니다, 피드백 부탁드립니다.
테스트에 대한 부분은 빠르게 채우겠습니다

BaeJunH0 committed:
feat : step 0, 기본 코드 설정 (#47)

Co-authored-by: BaeJunho <bjh4498@knu.ac.kr>

BaeJunH0 committed:
경북대 BE_배준호 3주차 과제 (1단계) (#72)

* feat : step 0, 기본 코드 설정

* docs : step 1 요구 사항 작성

* build : 빌드 파일 및 property 파일 작성

JPA에 대한 설정 추가

* feat : Entity 구현

기존 domain 클래스를 Entity 어노테이션을 통해 DB와 직접 매핑되는 Entity 클래스로 만들었다

* refactor : JPA 사용 Repository 구현

JpaRepository를 상속받는 interface 방식으로 repository 클래스를 변경하고, 이를 사용하는 service 클래스 역시 이에 맞게 바꾸었다

* fix : 더 이상 사용하지 않는 전처리 코드 삭제

* refactor : front 부분에서도 변화 적용

wish 객체에 수량을 추가한 부분, product 정보가 아닌 wishProduct 정보 전송하는 부분

* feat : 수량 수정 기능 추가

피드백에 따라 2주차 과제의 미구현 요구 사항 구현

* test : jpaDataTest 구현

* refactor : 피드백 사항 적용

* feat : separate user page and admin page

* chore : 더 이상 사용하지 않는 import 문 삭제

---------

Co-authored-by: BaeJunho <bjh4498@knu.ac.kr>

BaeJunH0 committed:
경북대 BE_배준호 3주차 과제(2단계) (#322)

* feat : step 0, 기본 코드 설정

* docs : step 1 요구 사항 작성

* build : 빌드 파일 및 property 파일 작성

JPA에 대한 설정 추가

* feat : Entity 구현

기존 domain 클래스를 Entity 어노테이션을 통해 DB와 직접 매핑되는 Entity 클래스로 만들었다

* refactor : JPA 사용 Repository 구현

JpaRepository를 상속받는 interface 방식으로 repository 클래스를 변경하고, 이를 사용하는 service 클래스 역시 이에 맞게 바꾸었다

* fix : 더 이상 사용하지 않는 전처리 코드 삭제

* refactor : front 부분에서도 변화 적용

wish 객체에 수량을 추가한 부분, product 정보가 아닌 wishProduct 정보 전송하는 부분

* feat : 수량 수정 기능 추가

피드백에 따라 2주차 과제의 미구현 요구 사항 구현

* test : jpaDataTest 구현

* refactor : 피드백 사항 적용

* feat : separate user page and admin page

* chore : 더 이상 사용하지 않는 import 문 삭제

* docs : 새로운 요구사항 반영

* style : DTO 클래스를 Response와 Request로 분리

* fix : 삭제된 파일

* refactor : change wishList's field to Other Entities

~id 형식이었던 wishList 클래스 필드 변수들을 실제 객체로 변경

* feat : 관리자 페이지와 유저 페이지 분리 및 위시리스트 페이지 구현

* style : 접근 제한자 지정

이전에 왜 접근 제한자를 지웠지? 아무튼 private 접근 제한자를 통해 외부에서 클래스 필드 변수 접근 못하게 바꾸었다

* fix : Transactional 어노테이션 삽입

CUD 동작에 대해 트랜잭션 수행되게 함 ( 안하면 update나 delete 시 오류가 나거나 바로 반영안되고 쿼리 캐싱됨 )

* chore : 메서드 명 변경

product 쪽에 대한 반영

* chore : 메서드 매개 변수를 userId에서 id로 변경

이 편이 좀 더 직관적인 것 같아서 변경

* chore : 백에서의 변화 반영

* style : change return value

userResponse를 반환하여 id를 사용할 수 있게 변경

* feat : implement delete logic

받는 값을 id가 포함된 userRes로 변경하고, 이를 통해 로직 간소화, 그리고 delete 로직에서 수량을 고려하도록 변경

---------

Co-authored-by: BaeJunho <bjh4498@knu.ac.kr>

BaeJunH0 committed:
경북대 BE_배준호 3주차 과제 (3단계) (#387)

* feat : step 0, 기본 코드 설정

* docs : step 1 요구 사항 작성

* build : 빌드 파일 및 property 파일 작성

JPA에 대한 설정 추가

* feat : Entity 구현

기존 domain 클래스를 Entity 어노테이션을 통해 DB와 직접 매핑되는 Entity 클래스로 만들었다

* refactor : JPA 사용 Repository 구현

JpaRepository를 상속받는 interface 방식으로 repository 클래스를 변경하고, 이를 사용하는 service 클래스 역시 이에 맞게 바꾸었다

* fix : 더 이상 사용하지 않는 전처리 코드 삭제

* refactor : front 부분에서도 변화 적용

wish 객체에 수량을 추가한 부분, product 정보가 아닌 wishProduct 정보 전송하는 부분

* feat : 수량 수정 기능 추가

피드백에 따라 2주차 과제의 미구현 요구 사항 구현

* test : jpaDataTest 구현

* refactor : 피드백 사항 적용

* feat : separate user page and admin page

* chore : 더 이상 사용하지 않는 import 문 삭제

* docs : 새로운 요구사항 반영

* style : DTO 클래스를 Response와 Request로 분리

* fix : 삭제된 파일

* refactor : change wishList's field to Other Entities

~id 형식이었던 wishList 클래스 필드 변수들을 실제 객체로 변경

* feat : 관리자 페이지와 유저 페이지 분리 및 위시리스트 페이지 구현

* style : 접근 제한자 지정

이전에 왜 접근 제한자를 지웠지? 아무튼 private 접근 제한자를 통해 외부에서 클래스 필드 변수 접근 못하게 바꾸었다

* fix : Transactional 어노테이션 삽입

CUD 동작에 대해 트랜잭션 수행되게 함 ( 안하면 update나 delete 시 오류가 나거나 바로 반영안되고 쿼리 캐싱됨 )

* chore : 메서드 명 변경

product 쪽에 대한 반영

* chore : 메서드 매개 변수를 userId에서 id로 변경

이 편이 좀 더 직관적인 것 같아서 변경

* chore : 백에서의 변화 반영

* style : change return value

userResponse를 반환하여 id를 사용할 수 있게 변경

* feat : implement delete logic

받는 값을 id가 포함된 userRes로 변경하고, 이를 통해 로직 간소화, 그리고 delete 로직에서 수량을 고려하도록 변경

* refactor : step 2.피드백 내용 반영

* docs : step 3 요구사항에 맞는 구현 내용 정리

* feat : 모든 데이터 페이지에 Page 기능 적용

---------

Co-authored-by: BaeJunho <bjh4498@knu.ac.kr>

Commit:
step 0 : 기본 코드 준비

Commit:
chore : delete no more use import

Commit:
fix : delete no more use class

Commit:
refactor : 필요없는 메서드 삭제

Commit:
docs : step 0 요구 사항 추가

Commit:
docs : 들여쓰기 문제 해결

Commit:
feat : impl category Entity class

Commit:
docs : step 1 요구사항 추가

Commit:
feat : 연관관계 구성 및 category 요소 추가 & product 관련 리펙터링

Product 클래스에 Category 사항을 적용하도록 리펙터링

Commit:
fix : 요구사항 변경에 따른 사양 수정

Commit:
docs : 요구사항 추가

Commit:
feat : 상품 조회 페이지에 Category 태그 추가

Commit:
fix : 위시 리스트 삭제 시 문제 해결

동일한 product를 한 번 삭제 이후 다시 삭제 시 생기던 문제를 해결했다 ( 변수 할당 문제였다 )

Commit:
feat : Category에 대한 CRUD API 구현

Category에 대한 Repository (JPA), Service, Controller를 구현하여 CRUD 동작을 수행하는 API 구현

BaeJunH0 committed:
Merge branch 'baejunh0' into step1

Commit:
chore : 공백제거

Commit:
buiid : 테스트 환경에서의 오류 제거를 위한 종속성 추가

Commit:
chore : 사용하지 않는 import문 제거

Commit:
chore : 사용하지 않는 메서드 제거

Commit:
refactor : 사용하지 않는 테스트 파일 제거

Commit:
feat : Entity 클래스에 대한 테스트 구현

Commit:
chore : 변수명 수정 (all -> users)

BaeJunH0 commented:
테스트 부분과 함께 정렬 방식을 View에서 선택하는 방식까지 구현해보겠습니다, 해당 부분 제외하고 피드백 해주세요!

lsm7179 commented on src/main/java/gift/Application.java:
```diff
@@ -1,11 +1,31 @@
 @SpringBootApplication
-public class Application{
+public class Application implements CommandLineRunner {
+    @Autowired
+    CategoryService categoryService;
+
     public static void main(String[] args) {
         SpringApplication.run(Application.class, args);
     }
+
+    @Override
+    public void run(String... strings) throws Exception {
+        String str = "교환권, 상품권, 뷰티, 패션, 식품, 리빙/도서, 레저/스포츠, 아티스트/캐릭터, 유아동/반려, 디지털/가전, 카카오프렌즈, 트렌드 선물, 백화점";
+        String[] split = str.split(", ");
+        for (String s : split) {
+            CategoryRequest category = new CategoryRequest(s);
+            categoryService.save(category);
+        }
+
+
+
+    }
```
jpa 초기 세팅하는 방법을 사용해보세요~
https://wildeveloperetrain.tistory.com/228

BaeJunH0 replied:
적용했습니다!

lsm7179 commented on src/main/java/gift/DTO/Product/ProductResponse.java:
```diff
@@ -1,35 +1,40 @@
     String name;
     int price;
     String imageUrl;
+    Category category;
```
`CategoryResponse`가 있으니 활용해보는건 어떨까요?

BaeJunH0 replied:
수정 했습니다! 제가 왜 이렇게 했는지... ㅎㅎ...

lsm7179 commented on src/main/java/gift/controller/CategoryController.java:
```diff
@@ -0,0 +1,47 @@
+
+import java.util.List;
+
+@Controller
```
`@RestController`가 더 적절해보이는데 어떤가요?

BaeJunH0 replied:
하하... 요즘 더워서 그런지 이상한 실수가 많네요... 정신 차리겠습니다!

lsm7179 commented on src/main/java/gift/controller/CategoryController.java:
```diff
@@ -0,0 +1,47 @@
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping("api/categories")
+    public ResponseEntity<List<CategoryResponse>> readCategory(){
+        List<CategoryResponse> categories = categoryService.findAll();
+
+        return new ResponseEntity<>(categories, HttpStatus.OK);
+    }
```
nit: 지난주에 사용했던 paging 처리도 하면 좋을거 같습니다.😁

BaeJunH0 replied:
적용했습니다!

lsm7179 commented on src/main/java/gift/controller/CategoryController.java:
```diff
@@ -0,0 +1,47 @@
+    public ResponseEntity<Void> deleteCategory(@PathVariable Long category_id){
+        categoryService.delete(category_id);
+
+        return new ResponseEntity<>(HttpStatus.OK);
```
[`HttpStatus.NO_CONTENT`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpStatus.html#NO_CONTENT) 
삭제는 상태코드로 no_content가 더 적절해보이네요.

BaeJunH0 replied:
수정했습니다~

lsm7179 commented on src/main/java/gift/domain/Category.java:
```diff
@@ -0,0 +1,32 @@
+@Entity
+@Table(name="categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
계속해서 엔티티 클래스에 반복되는 id
baseEntity를 도출해보세요.
https://devfunny.tistory.com/684

BaeJunH0 replied:
적용했습니다!

lsm7179 commented on src/main/java/gift/service/CategoryService.java:
```diff
@@ -0,0 +1,50 @@
+    }
+    @Transactional
+    public void save(CategoryRequest categoryRequest){
+        Category category = new Category(
+                categoryRequest.getName()
+        );
```
`categoryRequest`에서 카테고리를 직접 생성해보세요~

BaeJunH0 replied:
잘 이해가 안갑니다!

lsm7179 replied:
` Category category = categoryRequest .toCartegory()`메소드를 만들어서 categoryRequest클래스에서 객체생성하여 전달해보세요~

lsm7179 commented on src/main/java/gift/service/CategoryService.java:
```diff
@@ -0,0 +1,50 @@
+    }
+    @Transactional
+    public void update(Long id, CategoryRequest category){
+        Category savedCategory = categoryRepository.findById(id).orElseThrow();
```
예외처리는 안해도 될까요?

BaeJunH0 replied:
하겠습니다!

lsm7179 commented on src/main/java/gift/service/ProductService.java:
```diff
@@ -43,10 +47,13 @@ public ProductResponse readOneProduct(Long id){
      */
     @Transactional
     public void createProduct(ProductRequest product){
```
productRequest로 받아야 도메인객체가 아니다 라고 명확해질거같은데 어떻게 생각하시나요?

BaeJunH0 replied:
음... 저번에도 지적해주셨듯이 UserEntity와 User 보다는 User와 UserDTO 같은 형식이 알아보기 쉬울텐데, 또 까먹었나봅니다, 수정하겠습니다~

lsm7179 commented on src/main/java/gift/service/WishListService.java:
```diff
@@ -76,10 +76,10 @@ public void updateWishProduct(Long userId, Long productId, int count){
-    public void deleteWishProduct(Long userId, Long productId){
-        WishProduct wish = wishListRepository.findByUserIdAndProductId(userId, productId);
+    public void deleteWishProduct(Long wishId){
+        WishProduct wish = wishListRepository.findById(wishId).orElseThrow();
```
예외처리는 안해도 될까요?

BaeJunH0 replied:
하겠습니다!

lsm7179 commented:
> 테스트 부분과 함께 정렬 방식을 View에서 선택하는 방식까지 구현해보겠습니다, 해당 부분 제외하고 피드백 해주세요!

네! 따로 코멘트는 안하겠습니다.
혹시라도 테스트 코드 작성이 어렵거나 도움이 필요하면 언제든DM주세요!
일정조율도 가능하니 멘토링도 활용 해보세요.

lsm7179 reviewed (changes requested):
준호님 과제하느라 수고하셨습니다.
몇가지 코멘트 남겼습니다.
같이 반영해주세요~

Commit:
refactor : 전처리 방식 변경

Application 클래스에서 CommandLineRunner를 상속받아 처리하던 Category 전처리를 SQL 쿼리로 처리하는 방식으로 변경

Commit:
feat : 중복되는 id에 대한 선언을 상속을 통해 간략화

Commit:
feat : category 클래스에 대한 Paging 적용

Commit:
fix : 필드 변수 변경 (Entity -> DTO)

Commit:
style : DTO 객체 명 변경

기존의 방식은 Domain 객체와 DTO 객체를 알아보기 힘들어서, DTO 객체의 경우를 확실하게 표기 (Request나 Response 삽입)

Commit:
test : Repository 계층까지 테스트 작성

BaeJunH0 commented:
"categoryRequest에서 카테고리를 직접 생성해보세요~" 이 부분 외에는 피드백 및 테스팅 반영했습니다.
아직 상위 계층에서의 테스팅은 없고 Repository 계층까지만 구현했습니다...
step 2 진행하고 있겠습니다 ㅠㅠ

lsm7179 commented on src/main/java/gift/domain/BaseEntity.java:
```diff
@@ -0,0 +1,17 @@
+import jakarta.persistence.*;
+
+@MappedSuperclass
+public class BaseEntity {
```
👍

lsm7179 reviewed (approved):
준호님 수고하셨습니다~!
다음단계도 화이팅입니다.🔥
